src/txmempool.cpp

```diff
@@ -1309,11 +1309,15 @@ std::optional<std::string> CTxMemPool::CheckConflictTopology(const setEntries& d
     return std::nullopt;
 }
 
-util::Result<std::pair<std::vector<FeeFrac>, std::vector<FeeFrac>>> CTxMemPool::CalculateChunksForRBF(CAmount replacement_fees, int64_t replacement_vsize, const setEntries& direct_conflicts, const setEntries& all_conflicts)
+util::Result<std::pair<std::vector<FeeFrac>, std::vector<FeeFrac>>> CTxMemPool::ChangeSet::CalculateChunksForRBF()
 {
-    Assume(replacement_vsize > 0);
+    LOCK(m_pool->cs);
+    FeeFrac replacement_feerate{0, 0};
+    for (auto it : m_entry_vec) {
+        replacement_feerate += {it->GetModifiedFee(), it->GetTxSize()};
+    }
```

Comment:
after this loop, sanity check replacement_feerate isn't size 0?

Reply:
Ok, given new interface seems impossible to hit. Added another test case to `ComapreChunks` to just make sure it was well-defined:

```
diff --git a/src/test/rbf_tests.cpp b/src/test/rbf_tests.cpp
index c6d06c937c..d610661878 100644
--- a/src/test/rbf_tests.cpp
+++ b/src/test/rbf_tests.cpp
@@ -631,20 +631,32 @@ BOOST_AUTO_TEST_CASE(feerate_chunks_utilities)
 {
     // Sanity check the correctness of the feerate chunks comparison.
 
     // A strictly better case.
     std::vector<FeeFrac> old_chunks{{{950, 300}, {100, 100}}};
     std::vector<FeeFrac> new_chunks{{{1000, 300}, {50, 100}}};
 
     BOOST_CHECK(std::is_lt(CompareChunks(old_chunks, new_chunks)));
     BOOST_CHECK(std::is_gt(CompareChunks(new_chunks, old_chunks)));
 
+    // Empty comparisons
+    old_chunks = {{0, 0}};
+    new_chunks = {{1000, 300}, {0, 100}};
+
+    BOOST_CHECK(std::is_lt(CompareChunks(old_chunks, new_chunks)));
+    BOOST_CHECK(std::is_gt(CompareChunks(new_chunks, old_chunks)));
+
+    old_chunks = {{0, 0}};
+    new_chunks = {{0, 0}};
+
+    BOOST_CHECK(std::is_eq(CompareChunks(old_chunks, new_chunks)));
+
     // Incomparable diagrams
     old_chunks = {{950, 300}, {100, 100}};
     new_chunks = {{1000, 300}, {0, 100}};
 
     BOOST_CHECK(CompareChunks(old_chunks, new_chunks) == std::partial_ordering::unordered);
     BOOST_CHECK(CompareChunks(new_chunks, old_chunks) == std::partial_ordering::unordered);
 
     // Strictly better but smaller size.
     old_chunks = {{950, 300}, {100, 100}};
     new_chunks = {{1100, 300}};
```

Reply:
Maybe worth a separate PR to ensure the test coverage.